app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -197,8 +204,24 @@ public void onMenuItemClicked(View v) {
         menu.cancel();
         switch (v.getId()) {
             case R.id.menu_blockimg:
+                BLOCK_IMAGE_PREF = !BLOCK_IMAGE_PREF;
+                String blockImagePrefKey = this.getResources().getString(R.string.pref_key_performance_block_images);
+                PreferenceManager.getDefaultSharedPreferences(this)
+                        .edit()
+                        .putBoolean(blockImagePrefKey, BLOCK_IMAGE_PREF)
+                        .apply();
+
+                getBrowserFragment().reload();
```

Comment:
See onMenuBrowsingItemClicked() for suggestions. We also have a getVisibleBrowserFragment() on master branch for your convenience